### DISCUSSION TITLE
HDF deformation field file structure

### DISCUSSION THREAD

**Discussion by marianded** [ANTsX/ANTs#1556] [Q&A]
I am completely new to the subject. Therefore, it may be that my question is quite simple. But I didn't find anything about it in the documentation.

I have used the command antsRegistration with -a 1 to get the deformation field as HDF5 file. However, I do not understand all the values that are in the file. It contains a group "TransformGroup" which in turn contains groups "0", "1" and "2".

For example, "1" and "2" contains a dataset "TransformFixedParameters". But I don't understand what the values represent. 
In "1" they are:
2.0665789120387927
1.5655694997531708

and in "2" :
3000.0
3000.0
0.0
0.0
0.0013000000035390258
0.0013000000035390258
1.0
0.0
0.0
1.0

That the first two stand for values for the image size, I could still infer.

Can someone explain me the structure of this HDF5 file?

**Comment by cookpa** [comment]
Can you please include your ants version and the full command?

**Reply by marianded** [reply]
ANTs Version: 2.4.4.post12-g8cc4f8a

full command:
antsRegistration --verbose 1 --dimensionality 2 --float 0 --collapse-output-transforms 1 --output [ output,outputWarped.nii.gz,outputInverseWarped.nii.gz ] --interpolation Linear --use-histogram-matching 0 --winsorize-image-intensities [ 0.005,0.995 ] --initial-moving-transform [ PE-2010-00182-H_00_s1153_PM_HCB_Direction_Cropped_Original.nii,PE-2010-00182-H_00_s1153_PM_HCB_Direction_Cropped_30.nii,1 ] --transform Rigid[ 0.1 ] --metric MI[ PE-2010-00182-H_00_s1153_PM_HCB_Direction_Cropped_Original.nii,PE-2010-00182-H_00_s1153_PM_HCB_Direction_Cropped_30.nii,1,32,Regular,0.25 ] --convergence [ 1000x500x250x0,1e-6,10 ] --shrink-factors 12x8x4x2 --smoothing-sigmas 4x3x2x1vox --transform Affine[ 0.1 ] --metric MI[ PE-2010-00182-H_00_s1153_PM_HCB_Direction_Cropped_Original.nii,PE-2010-00182-H_00_s1153_PM_HCB_Direction_Cropped_30.nii,1,32,Regular,0.25 ] --convergence [ 1000x500x250x0,1e-6,10 ] --shrink-factors 12x8x4x2 --smoothing-sigmas 4x3x2x1vox --transform SyN[ 0.1,3,0 ] --metric MI[ PE-2010-00182-H_00_s1153_PM_HCB_Direction_Cropped_Original.nii,PE-2010-00182-H_00_s1153_PM_HCB_Direction_Cropped_30.nii,1,32] --convergence [ 100x100x70x50x0,1e-6,10 ] --shrink-factors 10x6x4x2x1 --smoothing-sigmas 5x3x2x1x0vox -a 1

**Comment by marianded** [comment]
Any help here?

**Reply by gdevenyi** [reply]
h5 transform files come from the ITK underlying IO system.

You should ask about those files here, https://discourse.itk.org/
